cmd/snap-bootstrap/cmd_initramfs_mounts.go

```diff
@@ -985,7 +985,14 @@ func (m *recoverModeStateMachine) setMountState(part, where string, err error) e
 }
 
 func (m *recoverModeStateMachine) setUnlockStateWithRunKey(partName string, unlockRes secboot.UnlockResult, err error) error {
-	part := m.degradedState.partition(partName)
+	if unlockRes.IsEncrypted {
+		m.isEncryptedDev = true
+	}
+	return m.degradedState.setUnlockStateWithRunKey(partName, unlockRes, err)
+}
+
+func (d *recoverDegradedState) setUnlockStateWithRunKey(partName string, unlockRes secboot.UnlockResult, err error) error {
```

Comment:
recoverDegradedState should be probably renamed to unlockState or something similar in a follow up